_posts/2017-05-10-polyglot-scientific-computing-with-xtensor.md

```diff
@@ -0,0 +1,195 @@
+---
+layout: post
+title: "Polyglot scientific computing with xtensor"
+date: 2017-05-010 20:20:40 +0100
+author: JohanMabille
+categories: [C++]
+excerpt_separator: <!--more-->
+---
+
```

Comment:
The `xtensor` expression system enables using tensor expressions backed by arbitrary data structures, from in-memory containers, data base calls or file system operations, but also the array types of scripting languages such as Python or Julia.

The [xtensor-python] and [xtensor-julia] projects respectively enable the use of numpy and julia array as tensor expressions. **This allows package authors to write their efficient C++ numerical libraries *once*, and create free bindings for Python and Julia and reach more users**, if they follow the best practices described in this post.